src/odemis/driver/test/technolution_test.py

```diff
@@ -0,0 +1,1460 @@
+        self.assertEqual(self.MPPC.cellTranslation.value,
+                         tuple(tuple((50, 50) for i in range(0, self.MPPC.shape[0])) for i in
+                               range(0, self.MPPC.shape[1])))
+
+        with self.assertRaises(ValueError):
+            self.MPPC.cellTranslation.value = tuple(tuple((50, -1) for i in range(0, self.MPPC.shape[0]))
+                                                    for i in range(0, self.MPPC.shape[1]))
+        self.assertEqual(self.MPPC.cellTranslation.value,
+                         tuple(tuple((50, 50) for i in range(0, self.MPPC.shape[0])) for i in
+                               range(0, self.MPPC.shape[1])))
+
+    def test_celldarkOffsetVA(self):
+        # Testing assigning different values. Which are chosen so that the value corresponds with the placement in
+        # the tuple
+        self.MPPC.cellDarkOffset.value = \
+            tuple(tuple(j for j in range(i, i + self.MPPC.shape[0]))
+                  for i in range(0, self.MPPC.shape[1] * self.MPPC.shape[0], self.MPPC.shape[0]))
+
+        self.assertEqual(
+                self.MPPC.cellDarkOffset.value,
+                tuple(tuple(j for j in range(i, i + self.MPPC.shape[0]))
+                      for i in range(0, self.MPPC.shape[1] * self.MPPC.shape[0], self.MPPC.shape[0]))
+        )
+
+        # Changing the digital gain back to something simple
+        self.MPPC.cellDarkOffset.value = tuple(
+                tuple(0 for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1]))
+
+        # Test missing rows
+        with self.assertRaises(ValueError):
+            self.MPPC.cellDarkOffset.value = tuple(tuple(0 for i in range(0, self.MPPC.shape[0] - 1))
+                                                   for i in range(0, self.MPPC.shape[1]))
+        self.assertEqual(self.MPPC.cellDarkOffset.value,
+                         tuple(tuple(0 for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1])))
+
+        # Test missing column
+        with self.assertRaises(ValueError):
+            self.MPPC.cellDarkOffset.value = tuple(tuple(0 for i in range(0, self.MPPC.shape[0]))
+                                                   for i in range(0, self.MPPC.shape[1] - 1))
+        self.assertEqual(self.MPPC.cellDarkOffset.value,
+                         tuple(tuple(0 for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1])))
+
+        # Test wrong type
+        with self.assertRaises(ValueError):
+            self.MPPC.cellDarkOffset.value = tuple(tuple(0.0 for i in range(0, self.MPPC.shape[0]))
+                                                   for i in range(0, self.MPPC.shape[1]))
+        self.assertEqual(self.MPPC.cellDarkOffset.value,
+                         tuple(tuple(0 for i in range(0, self.MPPC.shape[0])) for i in range(0, self.MPPC.shape[1])))
+
+        # Test minimum value setter
```

Comment:
You test that the minimum value allowed is set when a negative value is requested? Please rephrase this.